Lib/Protocols/IdFTPServer.pas

```diff
@@ -739,6 +739,20 @@   EIdFTPCannotBeNegative = class(EIdFTPServerException);
 
   TIdFTPServer = class;
 
+  TIdFTPServerInfo = class(TPersistent)

```

Comment:
A `CSID` response can contain additional name=value pairs (`os-dependant-fact`, `local-fact`, etc) so I would suggest adding a way for the user to provide those values as well.